File: leetCode/fukun/201-300/lc_263.js
var isUgly = function (num) {
    while (1) {
        let num2=num;
        num = num % 2 === 0 ? num / 2 : num;
        num = num % 3 === 0 ? num / 3 : num;
        num = num % 5 === 0 ? num / 5 : num;
        if(num===1)return true;
        if (num===num2)return false;
    }
};
console.log(isUgly(10));

File: leetCode/lwc/0201-0300/lc_263.java
package leetcode;

/**
 * @author lwc
 * @date 2019/11/26 20:43
 */
public class lc_263 {
    public boolean isUgly(int num) {
        if(num == 0) return false;
        while(num != 1){
            if(num % 2 == 0) num = num / 2;
            else if(num % 3 == 0) num = num / 3;
            else if(num % 5 == 0) num = num / 5;
            else return false;
        }
        return true;
    }
}
